otherlibs/fiber/pool.ml

```diff
@@ -1,17 +1,13 @@
 open Stdune
 open Core
-open Core.O
-
-type mvar =
-  | Done
-  | Task of (unit -> unit t)
 
 type status =
   | Open
   | Closed
 
-type t =
-  { mvar : mvar Mvar.t
+type nonrec t =
```

Comment:
Is there an invariant that this record needs to satisfy? If yes, it's worth documenting it here.

Reply:
(More generally, adding a top-level comment somewhere briefly explaining the main implementation idea would be nice.)

Reply:
Right, let me add some comments.